src/sync.h

```diff
@@ -260,6 +260,7 @@ inline MutexType* MaybeCheckNotHeld(MutexType* m) LOCKS_EXCLUDED(m) LOCK_RETURNE
     UniqueLock criticalblock2(MaybeCheckNotHeld(cs2), #cs2, __FILE__, __LINE__)
 #define TRY_LOCK(cs, name) UniqueLock name(MaybeCheckNotHeld(cs), #cs, __FILE__, __LINE__, true)
 #define WAIT_LOCK(cs, name) UniqueLock name(MaybeCheckNotHeld(cs), #cs, __FILE__, __LINE__)
+#define IN_PLACE_LOCK(cs) MaybeCheckNotHeld(cs), #cs, __FILE__, __LINE__
```

Comment:
In commit "validation: use a lock for CCheckQueueControl" (da5d55227e6aead8a9fbb85e967801cfcd4283ce)

Something like LOCK_ARGS might be a better name for this macro than than IN_PLACE_LOCK, because it expands to a list of 4 arguments, not a lock. Also, TRY_LOCK and WAIT_LOCK above could be deduplicated and written using LOCK_ARGS.

I don't think there is a close parallel between IN_PLACE_LOCK and std::in_place other than that they both get used as constructor arguments. IN_PLACE_LOCK isn't being used to help construct objects in place inside a larger std::any/std::optional/std::variant object like std::in_place is

Reply:
Done. I wasn't sure if `__LINE__` would work correctly when called from another macro, but it seems to do the right thing :)